src/main/java/com/gooddata/GoodData.java

```diff
@@ -192,10 +200,13 @@ protected GoodData(String hostname, String login, String password, int port, Str
         featureFlagService = new FeatureFlagService(restTemplate);
     }
 
-    private RestTemplate createRestTemplate(String hostname, HttpClient httpClient, int port, String protocol) {
-
+    static RestTemplate createRestTemplate(GoodDataEndpoint endpoint, HttpClient httpClient) {
```

Comment:
You can repeat it as many times you like, I won't write useless private method just because you ordered me to. If anyone needs private method he can write it.